File: backend/src/main/java/com/gymflow/pro/service/AuditService.java
package com.gymflow.pro.service;

import com.gymflow.pro.entity.User;
import jakarta.servlet.http.HttpServletRequest;

public interface AuditService {

    void log(User user, String action, String entityName, String entityId, String details, HttpServletRequest request);

    void log(String userName, String action, String entityName, String entityId, String details, HttpServletRequest request);
}

File: frontend/src/services/auditService.ts
import { api } from '@/services/api'
import type { Page, PageParams } from '@/types/common'
import type { AuditLog } from '@/types/audit'

interface AuditLogApiResponse {
  id: string
  userName: string
  action: string
  entityName: string
  entityId: string
  details?: string
  createdAt: string
}

export const auditService = {
  list: async (params?: PageParams): Promise<Page<AuditLog>> => {
    const { data } = await api.get<Page<AuditLogApiResponse>>('/audit-logs', { params })
    return {
      ...data,
      content: (data.content ?? []).map((log) => ({
        id: log.id,
        userName: log.userName,
        action: log.action,
        entity: log.entityName,
        entityId: log.entityId,
        details: log.details,
        timestamp: log.createdAt,
      })),
    }
  },
}
